packages/snjs/lib/application.ts

```diff
@@ -559,6 +560,28 @@ export class SNApplication {
     return compareVersions(userVersion, ProtocolVersion.V004) >= 0;
   }
 
+  public async getUserSubscription(): Promise<Subscription | undefined> {
+    try {
+      const response = await this.sessionManager.getSubscription();
+      if (!response.error && response.data) {
+        return (response as GetSubscriptionResponse).data!.subscription
+      }
+    } catch (e) {
+      return undefined;
+    }
```

Comment:
We want the errors to propagate:
```
if(response.error != null) throw new Error(error.message);
```

Reply:
And no need to wrap in try/catch. The error that happens in await will automatically be embedded in the error response part of the Promise.

Reply:
Done!